### PR TITLE
Use `config_for` for rubygems config

### PR DESCRIPTION
**Pull request by spk**
See https://gist.github.com/dhh/9afdbfd67e3c42c9f09e

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -11,29 +11,34 @@
 # you've limited to :test, :development, or :production.
 Bundler.require(*Rails.groups)
 
-$rubygems_config = YAML.load_file("config/rubygems.yml")[Rails.env].symbolize_keys
-HOST             = $rubygems_config[:host]
-
 module Gemcutter
   class Application < Rails::Application
+    # XXX: to remove when Rails updated to 4.2
```
Comment (arthurnn) [inline config/application.rb]: dont think the comment is necessary

File: config/application.rb [hunk 2]
```diff
@@ -11,29 +11,34 @@
...
+    def config_for(name, env = Rails.env)
+      YAML.load_file(Rails.root.join("config/#{name}.yml"))[env]
+    end
+
+    HOST = Application.config_for(:rubygems)['host']
```
Comment (arthurnn) [inline config/application.rb]: HOST should be on `Gemcutter` level only IMO

Comment (arthurnn) [inline config/application.rb]: So to access we do: `Gemcutter::HOST`

**Comment by arthurnn** [comment]
TBH we want to remove that config at all, but I guess we can do this before removing it completely

**Comment by spk** [comment]
@arthurnn thanks for your review

File: config/application.rb [hunk 3]
```diff
@@ -11,30 +11,35 @@
...
     config.middleware.use "Hostess"
-    config.middleware.insert_after "Hostess", "Redirector" if $rubygems_config[:redirector] && ENV["LOCAL"].nil?
+    if Application.config_for(:rubygems)['redirector'] && ENV["LOCAL"].nil?
+      config.middleware.insert_after "Hostess", "Redirector"
+    end
 
     unless Rails.env.maintenance?
-      config.action_mailer.default_url_options  = { :host => HOST }
-      config.action_mailer.delivery_method      = $rubygems_config[:delivery_method]
+      config.action_mailer.default_url_options  = { host: Application.config_for(:rubygems)['host'] }
```
Comment (arthurnn) [inline config/application.rb]: if we have a `default_url_options` on the only mailer we use, I wonder why this also needs to be here.

Comment (spk) [inline config/application.rb]: Do you mean we should put that on the mailer class ? Maybe on `config/environments/{development.rb,maintenance.rb,production.rb,recovery.rb,staging.rb,test.rb}` ?
See https://github.com/rubygems/rubygems.org/blob/master/config/environments/production.rb#L61

Comment (arthurnn) [inline config/application.rb]: it is on the mailer classes already https://github.com/spk/rubygems.org/blob/config_for/app/mailers/mailer.rb#L2 , what I am thinking is that we should remove it from there, and only leave it on the environments/*.rb files.

Comment (arthurnn) [inline config/application.rb]: If you could do that, would be great, also do it in a separate commit, so that change is isolated. thanks

File: config/application.rb [hunk 4]
```diff
@@ -11,30 +11,35 @@
 # you've limited to :test, :development, or :production.
 Bundler.require(*Rails.groups)
 
-$rubygems_config = YAML.load_file("config/rubygems.yml")[Rails.env].symbolize_keys
-HOST             = $rubygems_config[:host]
-
 module Gemcutter
   class Application < Rails::Application
+    def config_for(name, env = Rails.env)
+      YAML.load_file(Rails.root.join("config/#{name}.yml"))[env]
```
Comment (simi) [inline config/application.rb]: Isn't config file parsed everytime `config_for` is called?

Comment (arthurnn) [inline config/application.rb]: thats a really good point. we should have this as `Gemcutter.config`

**Comment by spk** [comment]
Thanks for your reviews :exclamation:

**Comment by simi** [comment]
I think `config_for` can be removed since it is used only once and assign memoized config hash directly into `Gemcutter.config`.

**Comment by spk** [comment]
I think it's not possible without duplication code because the `Rails.root` path is not already loaded.

**Comment by arthurnn** [comment]
> I think config_for can be removed since it is used only once and assign memoized config hash directly into Gemcutter.config.

I think we should keep the `config_for` method for now, it is a isolated and clean way to do it.

@spk Do you mind squashing your commits in one, so I can merge it?

thanks everyone.

**Comment by arthurnn** [comment]
thanks :heart:

**Comment by spk** [comment]
:beers:
